src/cli/commands/install.js

```diff
@@ -431,6 +431,25 @@ export class Install {
       this.scripts.setArtifacts(artifacts);
     }
 
+    if (Object.getPrototypeOf(this) === Install.prototype) {
+      const registryNames = Object.keys(registries);
+      for (let i = 0; i < registryNames.length; i++) {
+        const {filename} = registries[registryNames[i]];
+        const loc = path.join(this.config.cwd, filename);
+
+        // `fs.exists()` is deprecated since nodejs v8
+        try {
+          await fs.access(loc);
```

Comment:
No, it is called on line 421.
depRequests contains all the patterns from a package.json.
If it is empty and it is not an Add command then installation should stop without saving a lockfile.